### ISSUE TITLE
Occasional failures with parallel testing

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#1181]
@mdboom - I've started to see a few failures again very occasionally when testing in parallel using #1040 (since that is merged, I'm opening a new issue). Most of the time I  just get a few warnings:

```
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='/dev/null' mode='a' encoding='UTF-8'>
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='/dev/null' mode='a' encoding='UTF-8'>
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='/dev/null' mode='a' encoding='UTF-8'>
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='/dev/null' mode='a' encoding='UTF-8'>
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='/dev/null' mode='a' encoding='UTF-8'>
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='/dev/null' mode='a' encoding='UTF-8'>
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='/dev/null' mode='a' encoding='UTF-8'>
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='/dev/null' mode='a' 
etc. (happens as many times as processes)
```

but once I saw some failures: http://pastebin.com/8QqLeWYk - I only got this once, so it must be some race condition that only happens rarely, but I just thought I'd report them in case it helped. It might have been a fluke though - I haven't managed to reproduce it since.
